### ISSUE TITLE
Delete files after inactivity?

### ISSUE THREAD

**Issue by JamesTheAwesomeDude** [nokonoko/Uguu#11]
Would it be possible to implement a setting to delete a file once it's not been downloaded for a consecutive hour?

This could prove useful.

**Comment by nokonoko** [comment]
I'll look into it but that would mean it can't be served directly from Nginx unless I make some filthy hack that checks when it was accessed in the file system.

**Comment by avail** [comment]
A cron job that runs a php script?

**Comment by nokonoko** [comment]
Sure or just a bash script, however for any of it to work we need to know when it was last accessed by Nginx, so still applies to what I said above.

I guess I can mess around with it.

**Comment by avail** [comment]
json/sql(ite) database?

**Comment by JamesTheAwesomeDude** [comment]
Does not the filesystem provide read timestamps?

**Comment by avail** [comment]
No, only date created

**Comment by jackwilsdon** [comment]
One way would be to store the latest time downloaded when someone downloads a file, and then just check it the next time someone wants to download it. If an hour has passed since it was last downloaded then render the error page and delete the file in the background. This would also mean the file is deleted automatically after the 24 hours, even if accessed in the last hour.

**Comment by Battleroid** [comment]
Though it's shit, you could write something similar [like this](https://github.com/Battleroid/ppupload/blob/master/check.py). Just check for the ctime and remove if older than whatever by cronjob.

**Comment by JamesTheAwesomeDude** [comment]
@jackwilsdon The idea that you would intentionally deprive a user access to a file which does in fact exist and had been occupying disk space up to that very moment seems a little counterintuitive…

Better to find a way to delete files due to _in_ activity IMHO

**Comment by jackwilsdon** [comment]
@JamesTheAwesomeDude the issue with that is you'd need some form of script or process running in the background to handle that, whereas my suggestion is a passive one. I admit it's not the most efficient, and if someone only accesses a file once then it's there forever. Really the only *decent* way to do it involves a background process / script running periodically and sweeping up the expired files.

**Comment by geistesk** [comment]
This issue may be older but I want to suggest to use the `atime` (access time) of the file which is stored by the filesystem. The check.sh-script already uses `mtime` so why don't use the `atime`?
```bash
$ touch test
# wait some seconds
$ cat test
$ ls --full-time test
-rw-r--r-- 1 XXX XXX 0 2016-09-11 10:53:01.089956660 +0200 test
$ ls --full-time -u test # -u shows the atime
-rw-r--r-- 1 XXX XXX 0 2016-09-11 10:53:07.226623293 +0200 test
```

Please consider the `find` manpage,  which says that `-mtime` (and `-atime`) are for days and you want to use `-mmin` (and `-amin`). This issue was already addressed in #28.

**Comment by alucard0134** [comment]
(cough ls -ltu or ls -l --time=atime)
